### PR TITLE
CCMSG-635: Test DLQ logs with errant reporter

### PR DESCRIPTION
**Pull request by Prathibha-m**
## Problem
Change the Log4J configuration for each worker to send DLQ log messages to a different file, with its own rolling and file limit specification (see https://github.com/confluentinc/ce-kafka/blob/master/config/log4j.properties for examples of how Kafka sends different messages to different files)

## Solution
Test these log4j changes with IT
Relates to https://github.com/confluentinc/ce-kafka/pull/2826

##### Does this solution apply anywhere else?
- [ ] yes
- [ ] no

##### Testing done:
- [ ] Unit tests
- [ ] Integration tests
- [ ] System tests
- [ ] Manual tests

### CODE REVIEW

File: src/test/resources/log4j.properties [hunk 1]
```diff
@@ -28,3 +28,16 @@ log4j.logger.org.eclipse.jetty=ERROR
 log4j.logger.org.glassfish.jersey.internal=ERROR
 
 log4j.logger.io.confluent.connect.elasticsearch=DEBUG
+
+
+# Send the Errant Reporter logs to a DLQ file, rolling the file connect-dlq.log based on a Max Size of 1MB.
+#
+log4j.appender.dlqLogAppender=org.apache.log4j.RollingFileAppender
```
Comment (levzem) [inline src/test/resources/log4j.properties:35]: modifying log4j file isn't necessary

File: src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorIT.java [hunk 2]
```diff
@@ -138,4 +147,33 @@ public void testUpsert() throws Exception {
...
+  public void testBadRecordsInDLQ() throws Exception {
+    props.put(VALUE_CONVERTER_CLASS_CONFIG, StringConverter.class.getName());
+    props.put(BEHAVIOR_ON_MALFORMED_DOCS_CONFIG,"ignore");
+    props.put(DROP_INVALID_MESSAGE_CONFIG,"true");
+    props.put("errors.tolerance","all");
+    props.put("errors.deadletterqueue.topic.name","dlq_file_sink_02");
+    props.put("errors.deadletterqueue.topic.replication.factor", "1");
+    SinkTaskContext context = mock(SinkTaskContext.class);
+    ErrantRecordReporter reporter = mock(ErrantRecordReporter.class);
+    when(context.errantRecordReporter()).thenReturn(reporter);
```
Comment (levzem) [inline src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorIT.java:165]: dont need any of this

File: src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorIT.java [hunk 3]
```diff
@@ -138,4 +147,33 @@ public void testUpsert() throws Exception {
...
+   * does not wrap primitives in any way into a JSON document.
+   */
+  @Test
+  public void testBadRecordsInDLQ() throws Exception {
+    props.put(VALUE_CONVERTER_CLASS_CONFIG, StringConverter.class.getName());
+    props.put(BEHAVIOR_ON_MALFORMED_DOCS_CONFIG,"ignore");
+    props.put(DROP_INVALID_MESSAGE_CONFIG,"true");
+    props.put("errors.tolerance","all");
+    props.put("errors.deadletterqueue.topic.name","dlq_file_sink_02");
+    props.put("errors.deadletterqueue.topic.replication.factor", "1");
```
Comment (levzem) [inline src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorIT.java:162]: if these have config names, would be good to use them

File: src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorIT.java [hunk 4]
```diff
@@ -138,4 +147,33 @@ public void testUpsert() throws Exception {
...
+  /*
+   * Currently writing primitives to ES fails because ES expects a JSON document and the connector
+   * does not wrap primitives in any way into a JSON document.
+   */
+  @Test
+  public void testBadRecordsInDLQ() throws Exception {
+    props.put(VALUE_CONVERTER_CLASS_CONFIG, StringConverter.class.getName());
+    props.put(BEHAVIOR_ON_MALFORMED_DOCS_CONFIG,"ignore");
+    props.put(DROP_INVALID_MESSAGE_CONFIG,"true");
+    props.put("errors.tolerance","all");
```
Comment (levzem) [inline src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorIT.java:160]: is this necessary?

File: src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorIT.java [hunk 5]
```diff
@@ -138,4 +147,33 @@ public void testUpsert() throws Exception {
...
+    props.put("errors.deadletterqueue.topic.name","dlq_file_sink_02");
```
Comment (levzem) [inline src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorIT.java:161]:
```suggestion
    props.put("errors.deadletterqueue.topic.name","dlq");
```

File: src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorIT.java [hunk 6]
```diff
@@ -138,4 +147,33 @@ public void testUpsert() throws Exception {
...
+    connect.configureConnector(CONNECTOR_NAME, props);
+    
+    // wait for tasks to spin up
+    waitForConnectorToStart(CONNECTOR_NAME, TASKS_MAX);
+    
+    for (int i  = 0; i < NUM_RECORDS; i++) {
+      connect.kafka().produce(TOPIC, String.valueOf(i),  null);
+    }
+    
+    waitForRecords(0);
```
Comment (levzem) [inline src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorIT.java:177]: you should also consume from the dlq topic and make sure it actually contains the records you expect

File: src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorIT.java [hunk 7]
```diff
@@ -138,4 +147,33 @@ public void testUpsert() throws Exception {
       }
     }
   }
+  
+  /*
+   * Currently writing primitives to ES fails because ES expects a JSON document and the connector
+   * does not wrap primitives in any way into a JSON document.
+   */
```
Comment (levzem) [inline src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorIT.java:154]:
```suggestion
```

File: src/test/resources/log4j.properties (hunk 1 above)
Comment (Prathibha-m) [reply src/test/resources/log4j.properties:35]: This PR is primarily to test these changes https://github.com/confluentinc/ce-kafka/pull/2826 
Doesn't have to be merged here

**Pull request closed without merging**